### DISCUSSION TITLE
Weird goto_split:left goto_split:right behaviour

### DISCUSSION THREAD

**Discussion by rawkode** [ghostty-org/ghostty#2758] [Q&A] [answered]
With the setup below, I expected `goto_split:left` and `goto_split:right` to allow me to jump between the top two splits, but that's not how it works.

![Image](https://github.com/user-attachments/assets/2cc3d06d-3c19-4c4d-8259-29824bcd3122)

**Comment by 00-kat** [comment]
#2866[^1].

[^1]: You opened it first, but that's an issue while this is a discussion.

**Reply by rawkode** [reply]
https://github.com/ghostty-org/ghostty/issues/2756

**Reply by 00-kat** [reply]
Oh wow.

**Comment by rafikdraoui** [comment, marked as answer]
I haven't tried it yet to verify, but it seems like https://github.com/ghostty-org/ghostty/pull/8210 might be fixing this issue for GTK.

**Reply by jcollie** [reply]
Yes, give it a try when you can. A lot of work went into redoing the way splits (and navigation between them) works.

**Reply by Cybolic** [reply]
Has this been confirmed to work? I'm still seeing the split focus going by creation order instead of layout order in version 1.3.1 on Linux (NixOS nixos-unstable package).
